Add "How to Export to OpenAPI v3" to ytt

pivotaljohn opened the pull request.

jtigger committed:
Add "How to Export to OpenAPI v3" to ytt

cari-lynn commented on site/content/ytt/docs/latest/how-to-export-schema.md:
```diff
@@ -0,0 +1,159 @@
+
+The following are not yet supported in `ytt`:
+- `ytt` schema does not yet support declarative validations and thus does not produce such validations in an OpenAPI export.
+- there is no means yet to modify the [`info` section of the OpenAPI document](https://swagger.io/specification/#info-object) from within a `--data-values-schema-inspect`.
```
```suggestion
- there is no means yet to modify the [`info` section of the OpenAPI document](https://swagger.io/specification/#info-object) from within a `--data-values-schema-inspect`. There are defaults supplied, however we recommend that these fields are updated manually.
```

pivotaljohn replied:
> Document how to insert an OpenAPI components.schemas.dataValues into a kapp-controller Package (maybe in the kapp-controller docs?)

I think I failed to ping you on that PR: #301. I just do so, now.

pivotaljohn replied:
> add a @schema/desc annotation example somewhere so its easier to notice.

Love this and will...

pivotaljohn replied:
Done.

cari-lynn reviewed (approved):
Really like the way this is written, its an easy quick read that provides just enough detail. Also like that each key in the OpenAPI Doc is described, that is great since humans will need to read and interpret the doc (and its nice not to need to go to the spec).

A couple ideas of things that may be good to add. Not a blocker for this PR:
- ~~Document how to insert an OpenAPI  `components.schemas.dataValues` into a kapp-controller Package (maybe in the kapp-controller docs?)~~ Separate PR for this
- ~~add a `@schema/desc` annotation example somewhere so its easier to notice.~~ done

jtigger committed:
Add @schema/desc in the example

cari-lynn commented on site/content/ytt/docs/latest/how-to-export-schema.md:
```diff
@@ -0,0 +1,167 @@
+  host: ""
+  port: 5432
+
+#@schema/desc "Configuration for experimental/optional components; see documentation for more details."
```
👍

cari-lynn commented:
Awesome, looks great and ready to merge. I have a pending suggestion still if you want to take a look at that, but otherwise good to merge

pivotaljohn committed:
Clarify how OpenAPI `info` section fields are populated in docs

Co-authored-by: Cari <49043580+cari-lynn@users.noreply.github.com>

pivotaljohn commented:
> Awesome, looks great. I have a pending suggestion still if you want to take a look at that, but otherwise good to merge

Ooops! I swear I had incorporated that change... must have gotten lost in the shuffle.  Added.
